Admin Pull request

DimitriMaherovskyi opened the pull request.

Commit:
Admin pull request

DimitriMaherovskyi committed:
pull request commit

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/equiz.layoutController.js:
```diff
@@ -0,0 +1,27 @@
+﻿(function (angular) {
```
Use one file per component (https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#single-responsibility)

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/equiz.layoutController.js:
```diff
@@ -0,0 +1,27 @@
+﻿(function (angular) {
+    angular.module('equizModule').controller('LayoutController', function ($scope, $http, $sce) {
```
Use this style to define controllers: https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#getters

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/equiz.reviewController.js:
```diff
@@ -0,0 +1,192 @@
+﻿(function(angular) {
```
Use one file per object: https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#single-responsibility

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Views/Default/Index.cshtml:
```diff
@@ -1,16 +1,33 @@
-    ViewBag.Title = "Index";
-    Layout = "~/Views/Shared/Layout.cshtml";
+﻿@{
+    Layout = "~/Areas/Admin/Views/Shared/_Layout.cshtml";
```
Use general layout instead of defining a Layout within the Admin area

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Views/Shared/_Layout.cshtml:
```diff
@@ -0,0 +1,19 @@
+﻿@{ 
```
Do not define a layout page in the Admin area

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Views/web.config:
```diff
@@ -29,7 +29,7 @@
     <handlers>
       <remove name="BlockViewHandler"/>
-      <add name="BlockViewHandler" path="*" verb="*" preCondition="integratedMode" type="System.Web.HttpNotFoundHandler" />
+      <add name="BlockViewHandler" path="*cshtml" verb="*" preCondition="integratedMode" type="System.Web.HttpNotFoundHandler" />
```
What is the purpose of this line?

DimitriMaherovskyi replied:
MVC does not give access to HTML files in View folder.

vvoityshyn replied:
How do you load HTML templates?

DimitriMaherovskyi replied:
function LayoutController($scope, $http, $sce) {
        var vm = this;
        function GoToReview() {
            $http.get('/Areas/Admin/Views/Pages/Review.html')
            .then(function (response) {
                vm.review = $sce.trustAsHtml(response.data);
            });
        };

vvoityshyn replied:
Do not use the Views folder for AngularJS templates.
Put the templates into the Scripts folder instead (https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#application-structure-lift-principle).

vvoityshyn commented on eQuiz/modules/eQuiz.Web/eQuiz.Web.csproj:
```diff
@@ -89,19 +89,27 @@
     <Content Include="Scripts\app\shared\equiz.module.js" />
     <Content Include="Scripts\libs\angularjs\angular.js" />
     <Content Include="Scripts\libs\angularjs\angular.min.js" />
+    <Content Include="Scripts\libs\Underscore\underscore-min.js" />
```
Why do we need Underscore.js?

DimitriMaherovskyi replied:
It's the lib of methods, used to do table filter without writing huge amounts of code.

vvoityshyn replied:
As far as I see the following functions are used: 
_.contains()
_.without()

Consider to replace those functions with standard AngularJS analogues.

vvoityshyn replied:
The reason behind refusing underscorejs is that it does not fit AngularJS architecture.
In case we cannot avoid using it we need to employ angularjs-underscorejs adapter.

Commit:
Fixed structure and code convention

EugeneEX committed:
Fixed bug with group filter

DimitriMaherovskyi committed:
pull commit

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/Templates/Review.html:
```diff
@@ -0,0 +1,61 @@
+﻿<div ng-controller="ReviewController as rc">
```
1. Do not use the "Templates" folder, put the file into the root of the Scripts folder.
2. Use lower case for file's name - review.html

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/equiz.compileDirective.js:
```diff
@@ -0,0 +1,19 @@
+﻿(function (angular) {
```
Rename the file according to https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#naming

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/equiz.layoutController.js:
```diff
@@ -1,27 +1,17 @@
 ﻿(function (angular) {
```
Rename the file according to https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#naming

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/equiz.compileDirective.js:
```diff
@@ -0,0 +1,19 @@
+    Compile.$inject = ['$compile'];
+
+    function Compile($compile) {
+        return function (scope, element, attrs) {
```
Rename 'scope' with '$scope'

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/equiz.groupFilter.js:
```diff
@@ -0,0 +1,21 @@
+﻿(function (angular) {
```
Rename the file according to https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#naming

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/equiz.highlightFilter.js:
```diff
@@ -0,0 +1,15 @@
+﻿(function (angular) {
```
Rename the file according to https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#naming

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/equiz.reviewController.js:
```diff
@@ -4,15 +4,15 @@ angular.module('equizModule').controller('ReviewController', ReviewController);
     ReviewController.$inject = ['$scope', '$filter', 'reviewDataService'];
```
Rename the file according to https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#naming

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/equiz.startFromFilter.js:
```diff
@@ -0,0 +1,9 @@
+﻿(function(angular) {
```
Rename the file according to https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#naming

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Views/Default/Index.cshtml:
```diff
@@ -23,9 +23,13 @@
     <script src="~/Areas/Admin/Scripts/review/equiz.reviewController.js"></script>
+    <script src="~/Areas/Admin/Scripts/equiz.compileDirective.js"></script>
     <script src="~/Areas/Admin/Scripts/equiz.layoutController.js"></script>
     <script src="~/Scripts/libs/Underscore/underscore-min.js"></script>
```
Do we still need underscorejs?

EugeneEX committed:
Student profile page and controller communication service were added.

EugeneEX committed:
Files were renamed according to naming convenction. Template html's were renamed and placed into the root of the Scripts folder.

EugeneEX committed:
Bug with existence of unused child scope during directive's content recompilation was fixed.

Halyna-Posivnych committed:
deleting underscore and rewriting functions that uses it

EugeneEX committed:
Possible bug fix with ordering

EugeneEX committed:
Additional fix to ordering bug

DimitriMaherovskyi committed:
popUpWindow (not working)

Commit:
Modified review json

EugeneEX committed:
Bugs fixed:
1. links cursor was fixed;
2. default number of results per page is shown;
3. increased size of comments box;
4. cancel button was renamed to close;
5. row size in Quizzes and Students table was decreased plus pagination was added to Quizzes table;
6. recalculation of page number when searching was added;

EugeneEX committed:
Added comments and quizzes sorting (by date)

EugeneEX committed:
Ordering by date (Quizzes and Comments tables) was fixed

DemaPavlo committed:
Add quiz review page

DemaPavlo committed:
Merge branch 'master' of https://github.com/DimitriMaherovskyi/eQuiz.git

EugeneEX committed:
General comments were added.

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Controllers/old_DefaultController.cs:
```diff
@@ -0,0 +1,18 @@
+
+namespace eQuiz.Web.Areas.Admin.Controllers
+{
+    public class old_DefaultController : BaseController
```
Why do we need "old_" suffix?

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/compile.directive.js:
```diff
@@ -0,0 +1,19 @@
+﻿(function (angular) {
+    angular.module('equizModule').directive('compile', Compile);
```
Use the following convention:

angular
    .module('equizModule')
    .directive('compile', Compile);

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/layout.controller.js:
```diff
@@ -0,0 +1,36 @@
+﻿(function (angular) {
+    angular.module('equizModule').controller('LayoutController', LayoutController);
```
Use the following convention:

angular
    .module('equizModule')
    .controller('LayoutController', LayoutController);

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/layout.controller.js:
```diff
@@ -0,0 +1,36 @@
+        vm.GoToStudent = GoToStudent;
+        vm.GoToQuiz = GoToQuiz;
+
+        function GoToReview() {
```
Start function name with lower case

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/layout.controller.js:
```diff
@@ -0,0 +1,36 @@
+            });
+        };
+
+        function GoToStudent() {
```
Start function name with lower case

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/layout.controller.js:
```diff
@@ -0,0 +1,36 @@
+            });
+        };
+
+        function GoToQuiz() {
```
Start function name with lower case

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/group.filter.js:
```diff
@@ -0,0 +1,21 @@
+﻿(function (angular) {
+    angular.module('equizModule').filter('groupFilter', GroupFilter);
```
Use the following convention:

angular
.module('equizModule')
.filter('groupFilter', GroupFilter);

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/highlight.filter.js:
```diff
@@ -0,0 +1,15 @@
+﻿(function (angular) {
+    angular.module("equizModule").filter('highlight', HighlightFilter);
```
Use the following convention:

angular
.module("equizModule")
.filter('highlight', HighlightFilter);

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/review.controller.js:
```diff
@@ -0,0 +1,164 @@
+﻿(function(angular) {
+angular.module('equizModule').controller('ReviewController', ReviewController);
```
Use the following convention:

angular
.module('equizModule')
.controller('ReviewController', ReviewController);

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/review.controller.js:
```diff
@@ -0,0 +1,164 @@
+        vm.selectedGroup = [];
+    };
+
+    function GetUniquePropertyValues(arrayToCheck, propertyName) {
```
Start function name with lower case

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/review/start-from.filter.js:
```diff
@@ -0,0 +1,9 @@
+﻿(function(angular) {
+    angular.module('equizModule').filter('startFrom', function () {
```
Wrong syntax of filter declaration.
Use quizReview-data.service.js as an example

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/shared-values.service.js:
```diff
@@ -0,0 +1,15 @@
+﻿(function (angular) {
+    angular.module('equizModule').service('sharedProperties', function () {
```
Wrong syntax of a service declaration.
Use quizReview-data.service.js as an example

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/shared/popUp.js:
```diff
@@ -0,0 +1,25 @@
+﻿(function (angular) {
+    angular.module('equizModule')
```
Wrong syntax of a directive declaration.
Use quizReview-data.service.js as an example

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/student/student.controller.js:
```diff
@@ -0,0 +1,204 @@
+﻿(function (angular) {
+	angular.module('equizModule').controller('StudentController', StudentController);
```
Use the following convention:

angular
.module('equizModule')
.controller('StudentController', StudentController);

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Views/Default/Index.cshtml:
```diff
@@ -1,16 +1,45 @@
+    <script src="~/Areas/Admin/Scripts/review/highlight.filter.js"></script>
+    <script src="~/Areas/Admin/Scripts/review/start-from.filter.js"></script>
+
+    <script src="~/Areas/Admin/Scripts/quizReview/quizReview-data.service.js"></script>
```
rename with "quiz-review-data.service.js"

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Views/Default/Index.cshtml:
```diff
@@ -1,16 +1,45 @@
+    <script src="~/Areas/Admin/Scripts/shared-values.service.js"></script>
+    <script src="~/Areas/Admin/Scripts/student/student-data.service.js"></script>
+
+    <script src="~/Areas/Admin/Scripts/quizReview/quizReview.controller.js"></script>
```
Rename with "quizReview.controller.js"

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Views/Default/Index.cshtml:
```diff
@@ -1,16 +1,45 @@
+    <script src="~/Areas/Admin/Scripts/student/student.controller.js"></script>
+    <script src="~/Areas/Admin/Scripts/compile.directive.js"></script>
+    <script src="~/Areas/Admin/Scripts/layout.controller.js"></script>
+
```
Move styles to the "HeadStyles" section

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/shared/equiz.module.js:
```diff
@@ -0,0 +1,4 @@
+﻿(function(angular) {
```
Why do we need this file?

That mode is already created in the common area - "Scripts/app/shared/equiz.module.js"

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Views/Default/Index.cshtml:
```diff
@@ -1,16 +1,45 @@
+
+@section FootScripts
+{
+    <script src="~/Areas/Admin/Scripts/shared/equiz.module.js"></script>
```
Why do we need this file?

That mode is already created in the common area - "Scripts/app/shared/equiz.module.js"

EugeneEX committed:
Fixed bugs described in comments

DemaPavlo committed:
pull-reques: changed file names to "quiz-review"
quiz-review: Add button highlights, button cancel and close now work, checking checkboxes depending on user quiz answer

Commit:
DB insert + create

vvoityshyn commented on eQuiz/database/eQuiz.Database/create.sql:
```diff
@@ -1 +1,2 @@
-﻿-- scripts to create database structure
\ No newline at end of file
+﻿CREATE APPLICATION ROLE [create]
```
Why do we need an application role on DB level?

vvoityshyn commented on eQuiz/modules/eQuiz.Web/Areas/Admin/Scripts/quizReview/quiz-review.controller.js:
```diff
@@ -0,0 +1,78 @@
+﻿(function (angular) {
+    angular
+        .module("equizModule")
+        .controller('quizReviewController', quizReviewController);
```
Controller name should start from the upper case.

DimitriMaherovskyi committed:
Pull request (In progress)
